contracts/snowcones/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,804 @@
+    // but in exchange the refund on every call to nonReentrant will be lower in
+    // amount. Since refunds are capped to a percentage of the total
+    // transaction's gas, it is best to keep them low in cases like this one, to
+    // increase the likelihood of the full refund coming into effect.
+    uint256 private constant _NOT_ENTERED = 1;
+    uint256 private constant _ENTERED = 2;
+
+    uint256 private _status;
+
+    constructor () public {
+        _status = _NOT_ENTERED;
+    }
+
+    /**
+     * @dev Prevents a contract from calling itself, directly or indirectly.
+     * Calling a `nonReentrant` function from another `nonReentrant`
+     * function is not supported. It is possible to prevent this from happening
+     * by making the `nonReentrant` function external, and make it call a
+     * `private` function that does the actual work.
+     */
+    modifier nonReentrant() {
+        // On the first call to nonReentrant, _notEntered will be true
+        require(_status != _ENTERED, "ReentrancyGuard: reentrant call");
+
+        // Any calls to nonReentrant after this point will fail
+        _status = _ENTERED;
+
+        _;
+
+        // By storing the original value once again, a refund is triggered (see
+        // https://eips.ethereum.org/EIPS/eip-2200)
+        _status = _NOT_ENTERED;
+    }
+}
+
+contract ProtocolGovernance {
+    /// @notice governance address for the governance contract
+    address public governance;
+    address public pendingGovernance;
+    
+    /**
+     * @notice modifier to allow for easy gov only control over a function
+     */
+    modifier onlyGovernance() {
+        require(msg.sender == governance, "unauthorized sender (governance");
+        _;
+    }
+    
+    /**
+     * @notice Allows governance to change governance (for future upgradability)
+     * @param _governance new governance address to set
+     */
+    function setGovernance(address _governance) external onlyGovernance{
+        pendingGovernance = _governance;
+    }
+
+    /**
+     * @notice Allows pendingGovernance to accept their role as governance (protection pattern)
+     */
+    function acceptGovernance() external {
+        require(msg.sender == pendingGovernance, "acceptGovernance: !pendingGov");
+        governance = pendingGovernance;
+    }
+}
+
+contract Strategist {
+    /// @notice strategist address for the strategist contract
+    address public strategist;
+    address public pendingStrategist;
+
+    /**
+     * @notice modifier to allow for easy gov only control over a function
+     */
+    modifier onlyStrategist() {
+        require(msg.sender == strategist, "unauthorized sender (strategist)");
+        _;
+    }
+    
+    /**
+     * @notice Allows strategist to change strategist (for future upgradability)
+     * @param _strategist new strategist address to set
+     */
+    function setStrategist(address _strategist) external onlyStrategist {
```

Comment:
This setup excludes anyone but the Strategist which is not set from becoming a strategist. Should be modified in next deployment.